### PR TITLE
Adding newline(s) after header

### LINKED ISSUE #28: Feature request - Adding newline(s) after header
Thanks for a neat project, we started using this in a new project a few weeks ago and just like @maksnester requested in one of the issue threads we'd like to have a newline after our header. I made a PR.

### PR DESCRIPTION
**Pull request by opera-emoller**
- Updates the project to use the latest version of eslint and mocha.
- Adds support for a 3rd argument specifying number of newlines to add after the header.
- Review it without whitespace since a big chunk of the code got indented one level.

Fixes #28

### CODE REVIEW

**Comment by opera-emoller** [comment]
The travis check is failing when linting the node_modules folder which I though was always excluded. Do we need to bump the travis config to build with a newer node? I'm testing this on node 14.8.0 and the travis config is building with node 8 by the looks. Never used travis before so don't know if there are any other quirks I'm missing.

File: lib/rules/header.js [hunk 1]
```diff
@@ -5,7 +5,7 @@ var commentParser = require("../comment-parser");
 var os = require("os");
 
 function isPattern(object) {
-    return typeof object === "object" && object.hasOwnProperty("pattern");
+    return typeof object === "object" && Object.prototype.hasOwnProperty.call(object, "pattern");
```
Comment (opera-duanei) [inline lib/rules/header.js:8]: Why is this change needed? If it exist on the prototype it must also exist on the object?

File: lib/rules/header.js [hunk 2]
```diff
@@ -22,17 +22,34 @@ function excludeShebangs(comments) {
...
+// check if they are at the start of the file since that is already checked by
+// hasHeader().
 function getLeadingComments(context, node) {
-    return node.body.length ?
-        context.getComments(node.body[0]).leading :
-        context.getComments(node).leading;
+    var all = excludeShebangs(context.getSourceCode().getAllComments(node.body.length ? node.body[0] : node));
+    if (all[0].type.toLowerCase() === "block") {
+        return [all[0]];
+    } else {
```
Comment (opera-duanei) [inline lib/rules/header.js]: Would be nice to skip else to avoid indents :D

File: lib/rules/header.js [hunk 3]
```diff
@@ -22,17 +22,34 @@ function excludeShebangs(comments) {
...
+            if (!txt.match(/^(\r\n|\r|\n)$/)) {
+                break;
+            }
+        }
+        return all.slice(0, i);
+    }
 }
 
-function genCommentBody(commentType, textArray, eol) {
+function genCommentBody(commentType, textArray, eol, newlines) {
```
Comment (opera-duanei) [inline lib/rules/header.js]: Is `newlines` a number? Can it be renamed to numNewlines like where it is called?

File: lib/rules/header.js [hunk 4]
```diff
@@ -22,17 +22,34 @@ function excludeShebangs(comments) {
...
+    var eols = "";
+    for (var j = 0; j < newlines; ++j) {
```
Comment (opera-duanei) [inline lib/rules/header.js]: Is `eol` a string? Then 
```
const eols = eol.repeat(newlines)
```
would do the trick!

**Review by opera-duanei — APPROVED** [review]
:+1:

File: lib/rules/header.js (hunk 1 above)
Comment (opera-emoller) [reply lib/rules/header.js:8]: This project has this eslint rule enabled https://eslint.org/docs/rules/no-prototype-builtins

File: lib/rules/header.js (hunk 2 above)
Comment (opera-emoller) [reply lib/rules/header.js]: Ah yes. This PR evolved over time, should be possible to get rid of that now.

**Review by Stuk — APPROVED** [review]
This looks great, thanks!

**Comment by Stuk** [comment]
Published as 3.1.0

**Comment by DLehenbauer** [comment]
@emoller, @stuk - Works great, thank you!
